lib/carrierwave/uploader/download.rb

```diff
@@ -35,7 +36,10 @@ def http?
 
         def file
           if @file.blank?
-            @file = Kernel.open(@uri.to_s, "User-Agent" => "CarrierWave/#{CarrierWave::VERSION}")
+            headers = @remote_headers.
+              merge('User-Agent' => "CarrierWave/#{CarrierWave::VERSION}")
```

Comment:
Using `reverse_merge` can be useful when a user want to set specific User-Agent.